SIRShiny.rmd

```diff
@@ -0,0 +1,159 @@
+                   "Number of Days you want to see (Select -1 to see when the disease will go away)",
+                   min = -1,
+                   max = 200,
+                   step = 1, 
+                   value = 50),
+      sliderInput("Initialinf",
+                   "Initial People infected",
+                  min = 1,
+                  max = 100,
+                  step = 1, 
+                  value = 5),
+      sliderInput("Spread_Rate",
+                  "Spread Rate", 
+                  min = 0.01, 
+                  max = 1.00, 
+                  step = 0.01,
+                  value = 0.01),
+      sliderInput("Treat_Rate",
+                  "Treatment Rate", 
+                  min = 0.01, 
+                  max = 1.00, 
+                  step = 0.01,
+                  value = 0.01),
+      sliderInput("Susceptible_Rate",
+                  "Susceptible Rate",
+                  min = 0.01,
+                  max = 1,
+                  step = 0.01,
+                  value = 0.2)
+      
+    ),
+    
+    # Show a plot of the generated distribution
+    mainPanel(
+      plotOutput("distPlot")
+    )
+  )
+)
+
+# Define server logic required to draw a histogram
+server <- function(input, output) {
+  
+
+  
+  basic.model = function(n = 500, beta = 0.01, gamma = 0.7, S.rate = 0.2, I = 5, day = -1){
+    
+    # Calculate day 0 statistics
+    curr.day = 0
+    healthy = n - I
+    susceptible = round(S.rate * healthy)
+    infected = I
+    infected.next = round(I * beta * susceptible)
+    treat = round(gamma * I)
+    recovered = 0
+    
+    # Table storing S, I, R compartments over time
+    df = data_frame(Day = curr.day, Healthy = healthy, Susceptible = susceptible, Infected = infected, Recovered = recovered)
+    
+    # Predict S, I, R compartments by day
+    while(infected > 1){
+      # if number of days is specified, calculate up until that day
+      if(day != -1 & curr.day >= day)
+        break()
+      curr.day = curr.day + 1
+      healthy = healthy - infected.next
+      susceptible = round(S.rate * healthy)
+      infected = infected + infected.next - treat
+      infected.next = round(infected * beta * susceptible)
+      treat = round(gamma * infected)
+      recovered = recovered + treat
+      data = c(curr.day, healthy, susceptible, infected, recovered)
+      # data = c(curr.day, healthy, susceptible, infected, infected.next, treat, recovered)
+      df = rbind(df, data)
+    }
+    return(df)
+  }
+  
+  
+  model.plot = function(model){
+    ggplot(data  = model, aes(x = Day)) +
+      geom_line(aes(y = Healthy, color = "Healthy")) +
+      geom_line(aes(y = Infected, color = "Infected")) +
+      geom_line(aes(y = Recovered, color = "Recovered")) +
+      scale_color_discrete(name = "Compartment") +
+      labs(y = "Population", title = "Basic SIR Model") + 
+      theme_minimal()
+  }
+  
+  basic.model.rand = function(n = 500, beta = 0.01, gamma = 0.7, S.rate = 0.2, I = 5, t = 50){
```

Comment:
I think we just need a if-else statement in the prev function to keep our code clean and tidy rather than having another function with only 2 lines different